### PR TITLE
Refactor the agent.yaml configuration file #21067

### LINKED ISSUE #21067: Refactor the agent configuration file
### Is your feature request related to a problem?
 no 
### Describe the feature you would like.

1. The agent configuration file does not correspond to the module structure.
2. 'ignoredPluginNames' is difficult for users to understand.

#### Current agent.yml
```
applicationName: shardingsphere
ignoredPluginNames:
  - Jaeger
  - OpenTracing
  - Zipkin
  - OpenTelemetry
  - Logging
  - Prometheus

plugins:
  Prometheus:
    host:  "localhost"
    port: 9090
    props:
      JVM_INFORMATION_COLLECTOR_ENABLED : "true"
  Jaeger:
    host: "localhost"
    port: 5775
    props:
      SERVICE_NAME: "shardingsphere"
      JAEGER_SAMPLER_TYPE: "const"
      JAEGER_SAMPLER_PARAM: "1"
  Zipkin:
    host: "localhost"
    port: 9411
    props:
      SERVICE_NAME: "shardingsphere"
      URL_VERSION: "/api/v2/spans"
      SAMPLER_TYPE: "const"
      SAMPLER_PARAM: "1"
  OpenTracing:
    props:
      OPENTRACING_TRACER_CLASS_NAME: "org.apache.skywalking.apm.toolkit.opentracing.SkywalkingTracer"
  OpenTelemetry:
    props:
      otel.resource.attributes: "service.name=shardingsphere"
      otel.traces.exporter: "zipkin"
  Logging:
    props:
      LEVEL: "INFO"

```

####  Expected agent.yml
```
plugins:
    Logging:
      BaseLogging:
        props:
          xxx-xxx: xxx
    Metrics:
      Prometheus:
        host: "0.0.0.0"
        port: 9090
        props:
          xxx: xxx
    Tracing:
      Jaeger:
        host: "0.0.0.0"
        port: 5775
        props:
          xxx-xxx: xxx
      Zipkin:
        host: "0.0.0.0"
        port: 9411
        props:
          xxx-xxx: xxx
      OpenTracing:
        props:
          xxx-xxx: xxx
      OpenTelemetry:
        props:
          xxx-xxx: xxx
```

### PR DESCRIPTION
**Pull request by jiangML**
Fixes #21067.

Changes proposed in this pull request:
- Refactor: refactor the agent.yaml configuration file
- Adjust the format of the agent.yaml configuration parameter
- Update observability documentation
- Add metrics documentation
- Adjust SQLRouteEngineAdvice to fit SQLRouteEngine
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have triggered maven check: `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: shardingsphere-agent/shardingsphere-agent-distribution/src/main/resources/conf/agent.yaml [hunk 1]
```diff
@@ -15,43 +15,37 @@
...
+#        url-version: "/api/v2/spans"
+#        sampler-type: "const"
+#        sampler-param: "1"
+#    OpenTracing:
+#      props:
+#        opentracing-tracer-class-name: "org.apache.skywalking.apm.toolkit.opentracing.SkywalkingTracer"
+#    OpenTelemetry:
+#      props:
+#        otel-resource-attributes: "service.name=shardingsphere"
+#        otel-traces-exporter: "zipkin"
```
Comment (RaigorJiang) [inline shardingsphere-agent/shardingsphere-agent-distribution/src/main/resources/conf/agent.yaml:51]: Please keep a blank line at the end.

File: shardingsphere-test/shardingsphere-integration-agent-test/shardingsphere-integration-agent-test-plugins/shardingsphere-integration-agent-test-opentelemetry/src/test/resources/docker/proxy/conf/agent.yaml [hunk 2]
```diff
@@ -15,41 +15,40 @@
...
+#        sampler-type: "const"
+#        sampler-param: "1"
+#    OpenTracing:
+#      props:
+#        opentracing-tracer-class-name: "org.apache.skywalking.apm.toolkit.opentracing.SkywalkingTracer"
+    OpenTelemetry:
+      props:
+        otel-resource-attributes: "service.name=shardingsphere"
+        otel-traces-exporter: "zipkin"
+        otel-exporter-zipkin-endpoint: "http://zipkin.agent.tracing.opentelemetry.host:9411/api/v2/spans"
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-integration-agent-test/shardingsphere-integration-agent-test-plugins/shardingsphere-integration-agent-test-opentelemetry/src/test/resources/docker/proxy/conf/agent.yaml:54]: Please keep a blank line here.

File: docs/document/content/user-manual/shardingsphere-proxy/observability/_index.en.md [hunk 3]
```diff
@@ -40,66 +40,60 @@ tree
 * Configuration file
 
 `conf/agent.yaml` is used to manage agent configuration.
-Built-in plugins include Jaeger, OpenTracing, Zipkin, OpenTelemetry, Logging and Prometheus.
-When a plugin needs to be enabled, just remove the corresponding name in `ignoredPluginNames`.
+Built-in plugins include Jaeger, OpenTracing, Zipkin, OpenTelemetry, BaseLogging and Prometheus.
+No plugin are enabled by default. After configuring the corresponding plugin configuration, the corresponding plugin can be started.
```
Comment (RaigorJiang) [inline docs/document/content/user-manual/shardingsphere-proxy/observability/_index.en.md:44]: No plugin `is` enabled by default.      `is` for single plugin.

File: shardingsphere-agent/shardingsphere-agent-core/src/main/java/org/apache/shardingsphere/agent/core/plugin/AgentPluginLoader.java [hunk 4]
```diff
@@ -117,10 +117,20 @@ public void loadAllPlugins() throws IOException {
                 log.info("Loaded jar {}", each.getName());
             }
         }
-        loadPluginDefinitionServices(ignoredPluginNames, pointMap);
+        Set<String> pluginNames = getPluginNames();
```
Comment (RaigorJiang) [inline shardingsphere-agent/shardingsphere-agent-core/src/main/java/org/apache/shardingsphere/agent/core/plugin/AgentPluginLoader.java:120]: Could Set<String> be changed to Collection<String>?

File: shardingsphere-agent/shardingsphere-agent-distribution/src/main/resources/conf/agent.yaml (hunk 1 above)
Comment (jiangML) [reply shardingsphere-agent/shardingsphere-agent-distribution/src/main/resources/conf/agent.yaml:51]: ok

File: shardingsphere-test/shardingsphere-integration-agent-test/shardingsphere-integration-agent-test-plugins/shardingsphere-integration-agent-test-opentelemetry/src/test/resources/docker/proxy/conf/agent.yaml (hunk 2 above)
Comment (jiangML) [reply shardingsphere-test/shardingsphere-integration-agent-test/shardingsphere-integration-agent-test-plugins/shardingsphere-integration-agent-test-opentelemetry/src/test/resources/docker/proxy/conf/agent.yaml:54]: ok

File: shardingsphere-agent/shardingsphere-agent-core/src/main/java/org/apache/shardingsphere/agent/core/plugin/AgentPluginLoader.java (hunk 4 above)
Comment (jiangML) [reply shardingsphere-agent/shardingsphere-agent-core/src/main/java/org/apache/shardingsphere/agent/core/plugin/AgentPluginLoader.java:120]: Okay, I'll adjust it

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**